packages/compiler/init/init.ts

```diff
@@ -65,6 +65,13 @@ interface ScaffoldingConfig extends InitTemplate {
   normalizePackageName: () => (text: string, render: any) => string;
 }
 
+interface TemplatesUrl {
+  /** The original URL specified by the user. */
+  url: string;
+  /** The final URL after HTTP redirects. Populated when template is downloaded. */
+  finalUrl?: string;
```

Comment:
is it not something we could figure out ourself instead of requiring it? Is it common to have this pattern?

Reply:
The code is a little weird because I avoided significant refactoring, but this is figured out as follows: you pass in { url: X } and finalUrl is added to it so you get { url: X, finalUrl: Y }. This is all internal to the init module and the exported API takes url: string and "just works". Nothing new is required from caller.

The one change that has more impact is that the NodeHost will use the url of the response for the SourceFile path on readUrl instead of the url passed in. This is how init figures out there was a redirect, effectively.